code/modules/projectiles/gun.dm

```diff
@@ -59,131 +65,142 @@
 		O.emp_act(severity)
 
 /obj/item/weapon/gun/afterattack(atom/A as mob|obj|turf|area, mob/living/user as mob|obj, flag, params, struggle = 0)
-	if(flag)	return //we're placing gun on a table or in backpack
-	if(harm_labeled >= min_harm_label)
-		user << "<span class='warning'>A label sticks the trigger to the trigger guard!</span>" //Such a new feature, the player might not know what's wrong if it doesn't tell them.
+	if(flag) //We're placing gun on a table or in a backpack
 		return
-	if(istype(target, /obj/machinery/recharger) && istype(src, /obj/item/weapon/gun/energy))	return//Shouldnt flag take care of this?
-	if(user && user.client && user.client.gun_mode && !(A in target))
-		PreFire(A,user,params, "struggle" = struggle) //They're using the new gun system, locate what they're aiming at.
-	else
-		Fire(A,user,params, "struggle" = struggle) //Otherwise, fire normally.
+	if(harm_labeled >= min_harm_label) //Harm labelling can be used to sabotage the gun
+		user << "<span class='warning'>A label sticks the trigger to the trigger guard!</span>"
+		return
+	if(istype(target, /obj/machinery/recharger) && istype(src, /obj/item/weapon/gun/energy)) //Need to doublecheck this by commenting and testing

```

Comment:
@d3athrow or he could make rechargers return 1 for energy weapons and remove this whole section of code.